### PR TITLE
Update golangci-lint to 1.43.0

### PR DESCRIPTION
**Pull request by stmcginnis**
## What this PR does / why we need it

The version of golangci-lint we were using included a linter that had a
panic issue when running under go 1.17. This updates the version of
golangci-lint we use to a newer release that pulls in a set of updated
linters that fixes this, as well as adding improved lint checks.

One notable advantage of this is there is now an error emitted if the
deprecated ioutil package is used.

This update also uncovered a few other issues. Fixes for those are
included.

Go.sum files were out of date for some modules. We don't check that
after running out lint jobs, but running locally resulted in modified
files. So this picks up those changes too.

One thing of note is this also drops the gomnd lint check. Magic number
checks are useful, but in the majority of instances that this linter
pointed out, we just nolint the instance and ignore it. Rather than
enabling it globally and just ignoring it everywhere, this just disables
the global use of it.

## Details for the Release Notes (PLEASE PROVIDE)

```release-note
NONE
```

## Which issue(s) this PR fixes

Related: https://github.com/vmware-tanzu/tanzu-framework/issues/857

## Describe testing done for PR

Ran `make lint` locally and verified no errors reported and no more modified `go.sum` files.

### CODE REVIEW

File: addons/packages/harbor/2.2.3/test/go.sum [hunk 1]
```diff
@@ -1,3 +1,4 @@
+bazil.org/fuse v0.0.0-20160811212531-371fbbdaa898 h1:SC+c6A1qTFstO9qmB86mPV2IpYme/2ZoEQ0hrP+wo+Q=
```
Comment (karuppiah7890) [inline addons/packages/harbor/2.2.3/test/go.sum:1]: I just checked out to this PR using `gh pr checkout 2444` and ran `go mod tidy` inside `addons/packages/harbor/2.2.3/test` and saw that there was a diff in `git status` and `git diff`

Comment (karuppiah7890) [reply addons/packages/harbor/2.2.3/test/go.sum:1]: I think we can solve the problem of out of sync / outdated `go.mod` and `go.sum` files like this - https://github.com/vmware-tanzu/community-edition/pull/2167#issuecomment-935675031

Comment (karuppiah7890) [reply addons/packages/harbor/2.2.3/test/go.sum:1]: Created an issue to discuss this #2453

File: addons/packages/multus-cni/3.7.1/test/go.sum [hunk 2]
```diff
@@ -1,8 +1,10 @@
 github.com/davecgh/go-spew v1.1.0/go.mod h1:J7Y8YcW2NihsgmVo/mv3lAwl/skON4iLHjSsI+c5H38=
```
Comment (karuppiah7890) [inline addons/packages/multus-cni/3.7.1/test/go.sum:1]: Noticed the same issue as mentioned in https://github.com/vmware-tanzu/community-edition/pull/2444/files#r743377662

File: hack/tools/go.sum [hunk 3]
```diff
@@ -1,125 +1,267 @@
-4d63.com/gochecknoglobals v0.0.0-20201008074935-acfc0b28355a h1:wFEQiK85fRsEVF0CRrPAos5LoAryUsIX1kPW/WrIqFw=
-4d63.com/gochecknoglobals v0.0.0-20201008074935-acfc0b28355a/go.mod h1:wfdC5ZjKSPr7CybKEcgJhUOgeAQW1+7WcyK8OvUilfo=
+4d63.com/gochecknoglobals v0.1.0 h1:zeZSRqj5yCg28tCkIV/z/lWbwvNm5qnKVS15PI8nhD0=
```
Comment (karuppiah7890) [inline hack/tools/go.sum:1]: Noticed the same issue as mentioned in https://github.com/vmware-tanzu/community-edition/pull/2444/files#r743377662

**Review by karuppiah7890 — DISMISSED** [review]
LGTM! There were just some nitty gritty things I noticed, which is a common issue than something specific to this PR

**Review by chenliu1993 — DISMISSED** [review]

**Comment by chenliu1993** [comment]
Trying `go mod tidy` locally for multus. Multus refers to some outdated modules as this PR shows. Creating a issue [#2469](https://github.com/vmware-tanzu/community-edition/issues/2469) to track for multus.
So LGTM

**Review by davidvonthenen — DISMISSED** [review]

File: hack/runner/webhook/aws.go [hunk 4]
```diff
@@ -110,7 +109,7 @@ func deleteEc2InstanceByName(client *ec2.EC2, uniqueID string) error {
 			{
 				Name: aws.String("tag:Name"),
 				Values: []*string{
-					aws.String(strings.Join([]string{"*", uniqueID, "*"}, "")),
```
Comment (jpmcb) [inline hack/runner/webhook/aws.go:113]: Just curious, was this not valid from the linter? Seems perfectly reasonable to me (although it is using a literal string slice)

**Review by jpmcb — COMMENTED** [review]
LGTM!

File: hack/runner/webhook/aws.go (hunk 4 above)
Comment (jpmcb) [reply hack/runner/webhook/aws.go:113]: Huh! Interesting! Thanks for the thoughts!!

**Review by davidvonthenen — APPROVED** [review]

**Comment by dvonthenen** [comment]
@stmcginnis looks like there is a merge conflict now

**Pull request closed without merging**